test/unit/math/mix/fun/svd_U_test.cpp

```diff
@@ -0,0 +1,31 @@
+#include <test/unit/math/test_ad.hpp>
+TEST(MathMixMatFun, svd_U) {
+  auto f = [](const auto& x) { return stan::math::svd_U(x); };
+
+  stan::test::ad_tolerances tols;
+  tols.hessian_hessian_ = 1e-2;
+  tols.hessian_fvar_hessian_ = 1e-2;
+
+  Eigen::MatrixXd m00(0, 0);
+  stan::test::expect_ad(f, m00);
```

Comment:
To do varmat tests, for every `stan::test::expect_ad(f, ...)` do a `stan::test::expect_ad_matvar(f, ...)`.

And that's that! This should work for `singular_values`, `svd_U`, and `svd_V`. Lemme know if you have problems.